assets/src/data/model/entity-factory/base-entity.js

```diff
@@ -95,6 +95,8 @@ class BaseEntity {
 	get isClean() {
 		return this.saveState === SAVE_STATE.CLEAN;
 	}
+
+	static name = 'BaseEntity'
```

Comment:
Javascript sets the `constructor.name` property on a class automatically from whatever the name is given in the code ( eg. `class BaseEntity { }`).  But you can override this by explicitly declaring the name property on the class constructor.  If we don't do this then when a minifier changes something like `class BaseEntity` to `class a`, there's no way to know accurately elsewhere in the code what the name of the class is.  However, the minifier will leave _explicitly_ defined `name` constructor properties alone.

Reply:
kk kewl.. .I may have to do the same in some of the REM code